pkg/cloud/services/ec2/subnets.go

```diff
@@ -235,11 +256,21 @@ func (s *Service) describeVpcSubnets() (infrav1.Subnets, error) {
 }
 
 func (s *Service) createSubnet(sn *infrav1.SubnetSpec) (*infrav1.SubnetSpec, error) {
-	out, err := s.scope.EC2.CreateSubnet(&ec2.CreateSubnetInput{
+	subnet := &ec2.CreateSubnetInput{
 		VpcId:            aws.String(s.scope.VPC().ID),
 		CidrBlock:        aws.String(sn.CidrBlock),
 		AvailabilityZone: aws.String(sn.AvailabilityZone),
-	})
+	}
+
+	if s.scope.VPC().EnableIPv6 {
+		// The CIDR block has to have a /64 prefix size
+		// and use the prefix assigned to the VPC
+		ipv6CidrBlock := fmt.Sprintf("%s:%0x::/64", s.scope.VPC().Ipv6CidrBlock, sn.Ipv6CidrBlock)
```

Comment:
TODO: have to check what value is returning aws in  s.scope.VPC().Ipv6CidrBlock to create the right ipv6CidrBlack for the subnet